src/OrbitQt/orbitmainwindow.cpp

```diff
@@ -1270,4 +1268,29 @@ orbit_qt::TargetConfiguration OrbitMainWindow::ClearTargetConfiguration() {
         ->SetProcessListUpdateListener(nullptr);
   }
   return std::move(target_configuration_);
+}
+
+void OrbitMainWindow::ShowTooltip(std::string_view message) {
+  QToolTip::showText(QCursor::pos(),
+                     QString::fromUtf8(message.data(), static_cast<int>(message.size())), this);
+}
+
+void OrbitMainWindow::ShowSourceCode(const std::filesystem::path& file_path, size_t line_number) {
+  orbit_code_viewer::Dialog code_viewer_dialog{this};
+
+  code_viewer_dialog.SetEnableLineNumbers(true);
+  code_viewer_dialog.SetHighlightCurrentLine(true);
+
+  QFile source_code_file{QString::fromStdString(file_path.string())};
+
+  if (!source_code_file.open(QIODevice::ReadOnly)) {
+    ERROR("Could not open file %s", file_path.string());
+    return;
+  }
+
+  auto syntax_highlighter = std::make_unique<orbit_syntax_highlighter::Cpp>();
```

Comment:
Why it's unique_ptr and not just value?

Reply:
That's a requirement coming from Qt. This class inherits from QObject. And QObject is neither movable nor copyable. So I have to heap-allocate it. And then I can move it into the dialog object in the next line.

But in general you're right: A value would be much more desirable.

Reply:
Thanks for explanation!